Source/Utils/WarpXUtil.H

```diff
@@ -183,6 +183,111 @@ namespace WarpXUtilStr
+     * @return cont the split string
+     */
+    template <typename Container>
+    auto split (std::string const& instr, std::string const& separator,
+                  bool const trim = false, std::string const& trim_space = " \t")
+    {
+        Container cont;
+        std::size_t current = instr.find(separator);
+        std::size_t previous = 0;
+        while (current != std::string::npos) {
+            if (trim){
+                cont.push_back(amrex::trim(instr.substr(previous, current - previous),trim_space));}
+            else{
+                cont.push_back(instr.substr(previous, current - previous));}
+            previous = current + separator.size();
+            current = instr.find(separator, previous);
+        }
+        if (trim){
+            cont.push_back(amrex::trim(instr.substr(previous, current - previous),trim_space));}
+        else{
+            cont.push_back(instr.substr(previous, current - previous));}
+        return cont;
+    }
+
 }
 
+/**
+ * \brief This class is a parser for slices of the form i:j:k where i, j and k are integers
+ * representing respectively the starting point, the stopping point and the period.
+ */
+class SliceParser
+{
+    friend class MultiSliceParser;
+    int m_start = 0;
+    int m_stop = std::numeric_limits<int>::max();
+    int m_period = 1;
+    std::string m_separator = ":";
+
+    /**
+    * \brief Constructor of the SliceParser class.
+    *
+    * @param[in] instr an input string of the form "i:j:k", "i:j" or "k" where i, j and k are
+    * integers representing respectively the starting point, the stopping point and the period.
+    * Any of these integers may be omitted in which case it will be equal to their default value
+    * (0 for the starting point, std::numeric_limits<int>::max() for the stopping point and 1 for
+    * the period). For example SliceParser(":1000:") is equivalent to SliceParser("0:1000:1").
+    */
+    SliceParser (const std::string& instr);
+
+    /**
+    * \brief A method that returns true if the input integer is contained in the slice. (e.g. if
+    * the list is initialized with "300:500:100", this method returns true if and only if n is
+    * 300, 400 or 500). If the period is negative or 0, the function always returns false.
+    *
+    * @param[in] n the input integer
+    */
+    bool contains (int n);
+};
+
+/**
+ * \brief This class is a parser for multiple slices of the form x,y,z,... where x, y and z are
+ * slices of the form i:j:k, as defined in the SliceParser class. This class contains a vector of
+ * SliceParsers.
+ */
+class MultiSliceParser
+{
+    std::vector<SliceParser> m_slices;
+    std::string m_separator = ",";
+    bool m_activated = false;
+
+    /**
+    * \brief Constructor of the MultiSliceParser class.
+    *
+    * @param[in] instr an input string of the form "x,y,z,...". This will call the constructor of
+    * SliceParser using x, y and z as input arguments.
+    */
+    public:
```

Comment:
```suggestion
public:
```
This applies to all following members, and usually doesn't have this extra indentation.